src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2179,6 +2179,25 @@ private void EmitThisForMethodCall(Expression node)
             CompileAddress(node, -1);
         }
 
+        private static bool ShouldWritebackNode(Expression node)
+        {
+            if (!node.Type.GetTypeInfo().IsValueType)
+                return false;
+            switch (node.NodeType)
+            {
+                case ExpressionType.Parameter:
+                case ExpressionType.Call:
+                case ExpressionType.ArrayIndex:
```

Comment:
Ah.

Yes, I had thought it was more like `unbox.any` than `unbox` as a consequence of my thinking in C# terms.

The code remains correct though, as `Unbox` does not not need explicit write-back. And indeed whether we return `true` or `false` here the result will still be a path of calling `Compile(node)` and returning `null`.